xmhf/src/xmhf-core/xmhf-runtime/xmhf-eventhub/arch/x86_64/vmx/peh-x86_64vmx-main.c

```diff
@@ -992,7 +997,20 @@ u32 xmhf_parteventhub_arch_x86_64vmx_intercept_handler(VCPU *vcpu, struct regs *
 
 	//write updated VMCS back to CPU
 #ifndef __XMHF_VERIFICATION__
-	xmhf_baseplatform_arch_x86_64vmx_putVMCS(vcpu);
+	// [Superymk] Optimization to reduce trap overhead
+	if(vcpu->vmcs.info_vmexit_reason != VMX_VMEXIT_EPT_VIOLATION)
+		xmhf_baseplatform_arch_x86_64vmx_putVMCS(vcpu);
+	else
+	{
+		// Write back fewer fields in case of VMX_VMEXIT_EPT_VIOLATION
+		// [TODO][Issue 84] Remove the ugly VMCS indices
+		__vmx_vmwrite(0x681C, vcpu->vmcs.guest_RSP);
+		__vmx_vmwrite(0x681E, vcpu->vmcs.guest_RIP);
+		__vmx_vmwrite(0x4004, vcpu->vmcs.control_exception_bitmap);
+		__vmx_vmwrite(0x6820, vcpu->vmcs.guest_RFLAGS);
```

Comment:
I guess this optimization may work for some hypapp, but it does not work for TrustVisor. TrustVisor needs `vcpu->vmcs.control_EPT_pointer` to be vmwritten. Not sure whether other fields are also needed.